pyflagser/flagser.py

```diff
@@ -1,40 +1,150 @@
 """Implementation of the python API for the flagser C++ library."""
 
 import numpy as np
+
+from ._utils import _extract_unweighted_graph, _extract_weighted_graph
 from flagser_pybind import compute_homology, implemented_filtrations
 
 
-def flagser(flag_matrix, min_dimension=0, max_dimension=np.inf, directed=True,
-            filtration="max", coeff=2, approximation=-1):
+def flagser_unweighted(adjacency_matrix, min_dimension=0, max_dimension=np.inf,
+                       directed=True, coeff=2, approximation=None):
+    """Compute homology of a directed/undirected unweighted flag complex.
+
+    From an `adjacency_matrix` construct all cells forming its associated flag
+    complex and compute its homology.
+
+    Parameters
+    ----------
+    adjacency_matrix : 2d ndarray or scipy.sparse matrix, required
+        Adjacency matrix of a directed/undirected unweighted graph. It is
+        understood as a boolean matrix. Off-diagonal, ``0`` or ``False`` values
+        denote abstent edges while non-``0`` or ``True`` values denote edges
+        which are present. Diagonal values are ignored.
+
+    min_dimension : int, optional, default: ``0``
+        Minimum homology dimension to compute.
+
+    max_dimension : int or np.inf, optional, default: ``np.inf``
+        Maximum homology dimension to compute.
+
+    directed : bool, optional, default: ``True``
+        If ``True``, computes homology for the directed flad complex determined
+        by `adjacency_matrix`. If ``False``, computes homology for the
+        undirected flag complex obtained by considering all edges as
```

Comment:
Thanks! So it seems that this implies that if *either* entry (i, j) *or* entry (j, i) are non-zero, the undirected edge {i, j} exists. Correct? And thus it suffices to pass an upper triangular matrix. Do you think we could say this explicitly? E.g.: "If ``False``, computes homology for the undirected flag complex obtained by considering all edges as undirected, and it is sufficient (but not necessary) to pass an upper-triangular matrix."

Reply:
Exactly, thanks.